src/Extensions/ElementalPageExtension.php

```diff
@@ -43,7 +49,16 @@ public function getElementsForSearch()
                 /** @var ElementalArea $area */
                 $area = $this->owner->$key();
                 if ($area) {
-                    $output[] = strip_tags($area->forTemplate());
+                    // Only render the elemental blocks which are not excluded from search and don't pull from a page.
+                    foreach ($area->Elements() as $element) {
+                        if (!in_array($element->ClassName, Config::inst()->get(static::class, 'elements_excluded_from_search'))) {
+                            $forIndexing = $element->forTemplate();
```

Comment:
This always renders the element and then it decides whether it will throw it away based on the extension point.
Could it be reworked so excluded elements don't even render in the first place?

I can imagine something like adding `public function getIsIndexable(): bool` to the BaseElement and then each element can provide custom logic that would be used here to skip the rendering.

Reply:
>This always renders the element and then it decides whether it will throw it away based on the extension point.

The intention was that it excludes it if it's in the `elements_excluded_from_search` array, but then on a block-by-block basis, developers have even more control. So it might be a block which isn't excluded by class, but perhaps there is a rule where if some fields are not filled in it shouldn't be displayed (or indexed). So in that case the developer might choose to not index that specific block.
They can also choose to just modify what is indexed, in case the display of the block is not the same as what should be indexed for the block (I have at least one block where some fields are pulled from other sources, so I wouldn't want to index those fields as a part of this block).
If on the other hand the block's class is in the `elements_excluded_from_search` array, that if statement will return false and it will automatically be excluded.

I like the idea of a `getIsIndexable` method in principle provided it has an extension point in it to allow modifying that value (I use a lot of third-party blocks and want to define what is indexable for my project). But I think being able to modify _what_ is indexed per block is also important.

Reply:
Fair point about the modification, but basically at that point you're dealing with rendered markup, not structured data. So if you wanted to avoid e.g. an API call, that would be too late anyway with the above code.

There's an epic somewhere to provide a better content API that would allow for that granularity.

Reply:
I've made changes to allow per-block configuration both for whether a block is indexable in the first place, and if so, what content exactly will be indexed.
I think this is a much more sensible way of handling it. Please review.